[Doc][Improve] support chinese

JueLance opened the pull request:
<!--

Thank you for contributing to SeaTunnel! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/seatunnel/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

-->

### Purpose of this pull request

<!-- Describe the purpose of this pull request. For example: This pull request adds checkstyle plugin.-->


### Does this PR introduce _any_ user-facing change?
<!--
Note that it means *any* user-facing change including all aspects such as the documentation fix.
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If possible, please also clarify if this is a user-facing change compared to the released SeaTunnel versions or within the unreleased branches such as dev.
If no, write 'No'.
If you are adding/modifying connector documents, please follow our new specifications: https://github.com/apache/seatunnel/issues/4544.
-->


### How was this patch tested?
<!--
If tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check, and descendants can verify in the future.
If tests were not added, please describe why they were not added and/or why it was difficult to add.
If you are adding E2E test cases, maybe refer to https://github.com/apache/seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_mysql.conf, here is a good example.
-->


### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

Commit:
修正错别字

Commit:
[Doc][Improve]support chinese

Commit:
[Doc][Improve]support chinese

Commit:
[Doc][Improve]support chinese

Commit:
[Doc][Improve]support chinese

Commit:
[Doc][Improve]support chinese

Commit:
[Doc][Improve]support chinese

baicie commented:
run `mvn spotless:apply` to fix markdown style

Commit:
[Doc][Improve]format markdown document

Commit:
[Doc][Improve]format markdown document

baicie commented:
cant find connector-v2/Config-Encryption-Decryption.md in ./docs/zh/concept/JobEnvConfig.md

JueLance commented:
> cant find connector-v2/Config-Encryption-Decryption.md in ./docs/zh/concept/JobEnvConfig.md

Only the markdown in ./docs/zh/concept folder I have translated. The translation work of Config-Encryption-Decryption.md is still in progress, so I don't put it in ./docs/zh folder.

JueLance commented:
This pr is relation to issue #6366

Commit:
[Doc][Improve]add missing markdown file

Commit:
[Doc][Improve]add missing markdown file

TyrantLucifer reviewed (dismissed):
Thank you for your contribution. Unfortunately, you pull request has some empty files. Please double check and fix it.

JueLance commented:
> Thank you for your contribution. Unfortunately, you pull request has some empty files. Please double check and fix it.

I add empty file because github ci will check dead link. I follow previous commit from others add some empty file to pass it. If other contributor want continue to translate, just overwrite it.

davidzollo commented:
> > Thank you for your contribution. Unfortunately, you pull request has some empty files. Please double check and fix it.
> 
> I add empty file because github ci will check dead link. I follow previous commit from others add some empty file to pass it. If other contributor want continue to translate, just overwrite it.

good job, My suggestion is that you can add a message in the empty file. just like this,
"
ToDo(该文待翻译)
---------- ---------
我们正在补充和完善中文文档，如果您愿意贡献此文的翻译，请在 https://github.com/apache/seatunnel/issues/6378 下留言,  然后请提交一个翻译的 PR，一点一滴都是对 SeaTunnel 社区的贡献，感谢您的开源参与!  
"

Commit:
[Doc][Improve] removed empty files, modify link reference to en version.

JueLance commented:
> > > Thank you for your contribution. Unfortunately, you pull request has some empty files. Please double check and fix it.
> > 
> > 
> > I add empty file because github ci will check dead link. I follow previous commit from others add some empty file to pass it. If other contributor want continue to translate, just overwrite it.
> 
> good job, My suggestion is that you can add a message in the empty file. just like this, " ToDo(该文待翻译)
> 
> 我们正在补充和完善中文文档，如果您愿意贡献此文的翻译，请在 #6378 下留言, 然后请提交一个翻译的 PR，一点一滴都是对 SeaTunnel 社区的贡献，感谢您的开源参与! "

Thank you for your suggestion. I removed all empty file that I submit, and changed corresponding link to English version link. And I have already submit PR to #6378.

Commit:
[Doc][Improve] modify link reference to en version.

EricJoy2048 reviewed (dismissed):
LGTM, 

If Chinese document A is linked to other document B, but B has not been fully translated, we can link it to English document B first, so as to ensure the continuity of the document. After document B is translated, change the link address in document A to Chinese Document B.

ruanwenjun reviewed (commented):
Still exist empty file `docs/zh/other-engine/spark.md`

JueLance commented:
> Still exist empty file `docs/zh/other-engine/spark.md`

In English version, it's a empty file, too.

mosence commented on docs/zh/other-engine/spark.md:
You can write this EMPTY FILE with 
```markdown
ToDo  [docs/en/other-engine/spark.md](https://github.com/apache/seatunnel/blob/dev/docs/en/other-engine/spark.md)
```

mosence replied:
Or parent title link to ENGLISH file and DELETE this file.

JueLance committed:
Merge branch 'apache:dev' into dev

TyrantLucifer commented:
Overall LGTM, please waiting CI/CD

davidzollo committed:
Update spark.md

davidzollo reviewed (dismissed):
+1
LGTM

davidzollo committed:
Update spark.md

davidzollo committed:
Update spark.md

davidzollo committed:
Update spark.md

mosence reviewed (approved):
LGMT +1

JueLance committed:
Merge branch 'apache:dev' into dev

Commit:
[Doc][Improve] Add a new document for new developer to describe how to create a new connector base on SeaTunnel v2 connector API

Commit:
[Doc][Improve] add more content for empty file

Commit:
[Doc][Improve] document format

Commit:
[Doc][Improve] removed empty file
